src/assign.c

```diff
@@ -468,7 +468,7 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
       // strong error message for now.
     else if (TRUELENGTH(names) != oldtncol)
       // Use (long long) to cast R_xlen_t to a fixed type to robustly avoid -Wformat compiler warnings, see #5768, PRId64 didnt work
-      error(_("Internal error: selfrefnames is ok but tl names [%ld] != tl [%ld]"), TRUELENGTH(names), oldtncol);  // # nocov
+      error(_("Internal error: selfrefnames is ok but tl names [%ld] != tl [%d]"), TRUELENGTH(names), oldtncol);  // # nocov
```

Comment:
oh, interesting, does that mean TRUELENGTH(names) and TRUELENGTH(dt) have different types??

I noticed the Wformat error only hit this line once & was wondering why.

Reply:
It appears to be like that. I didn't try to understand that.